src/extrapolation/filled.jl

```diff
@@ -54,6 +54,17 @@ end
     return ret
 end
 
+@inline function Interpolations.gradient(etp::FilledExtrapolation{T}, x::Vararg{Number,N}) where {T,N}
+    itp = parent(etp)
+    if checkbounds(Bool, itp, x...)
+        gradient(itp, x...)
+    else
+        e = eltype(itp)
+        ptype = promote_type(e, mapreduce(typeof, promote_type, x))
+        return SVector{ndims(etp), ptype}(zeros(ptype, ndims(etp)))
```

Comment:
`zeros(ptype, ndims(etp))` is unnecessary intermediate memory allocation, let's use tuple here.

```suggestion
        ptype = promote_type(eltype(itp), map(typeof, x)...)
        return SVector{ndims(etp), ptype}(ntuple(_->zero(ptype), ndims(etp)))
```

```julia
julia> etp = extrapolate(interpolate([1, 2, 3], BSpline(Linear())), 0);

# before
julia> @btime Interpolations.gradient($etp, -0.5f0);
  35.995 ns (1 allocation: 96 bytes)
# after
julia> @btime Interpolations.gradient($etp, -0.5f0);
  0.038 ns (0 allocations: 0 bytes)
```